### PR TITLE
Run container as non-root user

### PR DESCRIPTION
**Pull request by srjayep**
**Running container api-gateway service  as non-root user**
Reference : CIS 4.1 section. https://benchmarks.cisecurity.org/tools2/docker/CIS_Docker_1.11.0_Benchmark_v1.0.0.pdf 

By default docker containers run as root  and as user root inside the container.  A docker container running as root has full control of the host system. In order to run the docker container image as a non-root user and api-gateway service to continue to user port 80, following modifications are done to dockerfile and init.sh

**#install sudo in docker image**
geoip-dev sudo iptables

**#Add user and group (uid and gid 1000 for consistency)**
 RUN adduser -S nginx-api-gateway -u 1000 \
   && addgroup -S nginx-api-gateway -g 1000

**#apply non-root user to directories and files.** 
 RUN mkdir -p /usr/local/api-gateway \
   && chown -R nginx-api-gateway /etc/api-gateway /var/log/api-gateway /usr/local \ 
   && chmod 755 -R /etc/api-gateway /var/log/api-gateway /usr/local \
   && chmod 4755 /bin/busybox \
   && echo "nginx-api-gateway ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers

**#USER instruction to specify a non-root user for container to run as**
USER nginx-api-gateway

After the above changes, the below command shows the container is running with user nginx-api-gateway  If it returns blank instead of username means, the container is running as root. 

**Example run**
docker ps --quiet | xargs docker inspect --format '{{ .Id }}: User={{.Config.User}}'
919da15af3df3f12d966659a37898b2dd4cbb7837f2989749adf14f6d430284c: **User=nginx-api-gateway**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -41,10 +41,11 @@ RUN echo " ... adding throttling support with ZMQ and CZMQ" \
          && make && make install \
          && apk del automake autoconf \
          && rm -rf /tmp/zeromq* && rm -rf /tmp/czmq* \
-         && rm -rf /var/cache/apk/*
+         && rm -rf /var/cache/apk/* 
 
 # openresty build
 ENV OPENRESTY_VERSION=1.9.7.3 \
+    NAXSI_VERSION=0.53-2 \
```
Comment (ddragosd) [inline Dockerfile]: NAXSI has been recently taken out due to license incompatibility.See latest commit.

File: Dockerfile [hunk 2]
```diff
@@ -270,7 +275,7 @@ RUN echo " ... installing api-gateway-aws ..." \
     && rm -rf /var/cache/apk/* \
     && rm -rf /tmp/api-gateway
 
-ENV REQUEST_VALIDATION_VERSION 1.2.0
+ENV REQUEST_VALIDATION_VERSION 1.1.1
```
Comment (ddragosd) [inline Dockerfile]: Please fix the module version to the latest release.

File: Dockerfile [hunk 3]
```diff
@@ -304,7 +309,7 @@ RUN echo " ... installing api-gateway-async-logger ..." \
     && rm -rf /var/cache/apk/* \
     && rm -rf /tmp/api-gateway
 
-ENV ZMQ_ADAPTOR_VERSION 0.2.1
+ENV ZMQ_ADAPTOR_VERSION 0.1.1
```
Comment (ddragosd) [inline Dockerfile]: Please fix the module version to the latest release.

**Review by ddragosd — CHANGES_REQUESTED** [review]

File: api-gateway-config/conf.d/default.conf [hunk 4]
```diff
@@ -21,7 +21,7 @@
 # *
 # */
 server {
-    listen 80 default_server;
+    listen 8080 default_server;
```
Comment (ddragosd) [inline api-gateway-config/conf.d/default.conf]: do we need to switch to port `8080` instead of `80` ?

**Review by ddragosd — APPROVED** [review]

**Comment by ddragosd** [comment]
@jkt628 I wanted to get your feedback on this PR before merging it as it may impact a little your WIP.

**Comment by jkt628** [comment]
thanks, @ddragosd , i do need to investigate this with @MikeWillCook.
https://github.com/adobe-community/issues-ethos/issues/1945

**Comment by adrocknaphobia** [comment]
@jkt628 @MikeWillCook We're going to need to merge this today. Have you had a had a chance to review?

**Comment by jkt628** [comment]
no, moving into holidays which is very busy for Analytics, just merge it and i'll deal with the fall out later.

**Pull request merged**